Sources/Filters/General/ContourTriangulator/index.js

```diff
@@ -0,0 +1,2289 @@
+      return 0.0; // to avoid false positive warning about uninitialized value
+    }
+
+    // Pull out the points
+    for (let k = 0; k < n; k++) {
+      // double* p = &pp[3 * k];
+      // points.getPoint(outerPoly[k], p);
+      pp[3*k] = outerPoly[k];
+      pp[3*k+1] = outerPoly[k+1];
+      pp[3*k+2] = outerPoly[k+2];
+    }
+
+    // Find the bounding box and tolerance for the polygon
+    return vtkCCSPolygonBounds(outerPoly, points, bounds) * (VTK_CCS_POLYGON_TOLERANCE * VTK_CCS_POLYGON_TOLERANCE);
+  }
+
+  // ---------------------------------------------------
+  // Check for polygons within polygons.  Group the polygons
+  // if they are within each other.  Reverse the sense of
+  // the interior "hole" polygons.  A hole within a hole
+  // will be reversed twice and will become its own group.
+
+  function vtkCCSMakeHoleyPolys(newPolys, points, polyGroups, polyEdges, originalEdges, normal, oriented) {
+    const numNewPolys = newPolys.length;
+    if (numNewPolys <= 1) {
+      return;
+    }
+
+    // Use bit arrays to keep track of inner polys
+    const polyReversed = new vtkCCSBitArray();
+    const innerPolys = new vtkCCSBitArray();
+
+    // GroupCount is an array only needed for unoriented polys
+    // TODO: Check
+    // size_t* groupCount = null;
+    // if (!oriented) {
+    //   groupCount = new size_t[numNewPolys];
+    //   std::fill(groupCount, groupCount + numNewPolys, 0);
+    // }
+    let groupCount;
+    if (!oriented) {
+      groupCount = [];
+      groupCount.length = numNewPolys;
+      groupCount.fill(0);
+    }
+
+
+    // Find the maximum poly size
+    let nmax = 1;
+    for (let kk = 0; kk < numNewPolys; kk++) {
+      let n = newPolys[kk].length;
+      if (n > nmax) {
+        nmax = n;
+      }
+    }
+
+    // These are some values needed for poly-in-poly checks
+    // TODO: Check
+    // double* pp = new double[3 * nmax];
+    const pp = new Float64Array(3*nmax);
+    let bounds = [0.0, 0.0, 0.0, 0.0, 0.0, 0.0];
+    let tol2;
+
+    // Go through all polys
+    for (let i = 0; i < numNewPolys; i++) {
+      let n = newPolys[i].length;
+
+      if (n < 3) {
+        continue;
+      }
+
+      // Check if poly is reversed
+      const { isNormalNotZero, sense } = vtkCCSCheckPolygonSense(newPolys[i], points, normal);
+      if (isNormalNotZero) {
+        polyReversed.set(i, !sense);
+      }
+
+      // Precompute some values needed for poly-in-poly checks
+      tol2 = vtkCCSPrepareForPolyInPoly(newPolys[i], points, pp, bounds);
+
+      // Look for polygons inside of this one
+      for (let j = 0; j < numNewPolys; j++) {
+        if (j != i && newPolys[j].length >= 3) {
+          // Make sure polygon i is not in polygon j
+          let pg = polyGroups[j];
+          // TODO: Check
+          // if (std::find(pg.begin(), pg.end(), i) == pg.end()) {
+          if (!pg.includes(i)) {
+            if (vtkCCSPolyInPoly(newPolys[i], newPolys[j], points, normal, pp, bounds, tol2)) {
```

Comment:
if `vtkCCSPolyInPoly` relies on `pp`'size, then you might have to call `pp.subArray(0, 3*n)`